src/components/App.jsx

```diff
@@ -127,34 +128,40 @@ export default function App() {
           <rb.Alert variant="danger">No connection to backend: {connectionError}.</rb.Alert>
         ) : (
           <Routes>
-            <Route exact path="/" element={<Wallets startWallet={startWallet} stopWallet={stopWallet} />} />
-            <Route
-              path="create-wallet"
-              element={<CreateWallet currentWallet={currentWallet} startWallet={startWallet} />}
-            />
+            <Route element={<Layout />}>
+              <Route exact path="/" element={<Wallets startWallet={startWallet} stopWallet={stopWallet} />} />
+              <Route
+                path="create-wallet"
+                element={<CreateWallet currentWallet={currentWallet} startWallet={startWallet} />}
+              />
+              {currentWallet && (
+                <>
+                  <Route
+                    path="send"
+                    element={<Send makerRunning={makerRunning} coinjoinInProcess={coinjoinInProcess} />}
+                  />
+                  <Route
+                    path="earn"
+                    element={
+                      <Earn
+                        currentWallet={currentWallet}
+                        coinjoinInProcess={coinjoinInProcess}
+                        makerRunning={makerRunning}
+                      />
+                    }
+                  />
+                  <Route path="receive" element={<Receive currentWallet={currentWallet} />} />
+                  <Route path="settings" element={<Settings currentWallet={currentWallet} />} />
+                </>
+              )}
+            </Route>
             {currentWallet && (
-              <>
+              <Route element={<Layout variant={settings.useAdvancedWalletMode ? 'wide' : 'narrow'} />}>
                 <Route
                   path="wallet"
                   element={settings.useAdvancedWalletMode ? <CurrentWalletAdvanced /> : <CurrentWalletMagic />}
                 />
-                <Route
-                  path="send"
-                  element={<Send makerRunning={makerRunning} coinjoinInProcess={coinjoinInProcess} />}
-                />
-                <Route
-                  path="earn"
-                  element={
-                    <Earn
-                      currentWallet={currentWallet}
-                      coinjoinInProcess={coinjoinInProcess}
-                      makerRunning={makerRunning}
-                    />
-                  }
-                />
-                <Route path="receive" element={<Receive currentWallet={currentWallet} />} />
-                <Route path="settings" element={<Settings currentWallet={currentWallet} />} />
-              </>
+              </Route>
             )}
```

Comment:
Do you think
```jsx
{currentWallet && !settings.useAdvancedWalletMode && (
  <Route element={<Layout variant='narrow'/>}>
    <Route path="wallet" element={<CurrentWalletMagic />} />
  </Route>
)}
{currentWallet && settings.useAdvancedWalletMode && (
  <Route element={<Layout variant='wide'/>}>
    <Route path="wallet" element={<CurrentWalletAdvanced />} />
  </Route>
)}
```

is more readable? Or does it hide the fact that the same path is used by both elements?

Reply:
100 times better! 👍